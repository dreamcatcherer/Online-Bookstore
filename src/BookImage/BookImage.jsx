import React from "react";
import styles from "./BookImage.module.css"

const BookImage = ({url, title}) => {
  return (
    <div className = {styles.imageStyle}>
      <img src={url} alt={title} />
    </div>
  );
};

export default BookImage;