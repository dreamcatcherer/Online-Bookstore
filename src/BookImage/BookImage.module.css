.bookCoverStyle {
  width: 320px;
  height: 320px;
  margin: 10px;
  border-radius: 15px;
  background-color: #ffb3b3;
  box-shadow: 5px 5px 10px #ffcccc;
  border: 5px solid #ff9999;
}

.bookCoverStyle img {
  padding: 10px;
  max-width: 90%;
  max-height: 90%;
  margin: auto;
  border-radius: 5%;
}
