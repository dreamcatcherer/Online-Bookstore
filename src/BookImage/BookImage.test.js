import React from "react";
import { render } from "@testing-library/react";
import BookImage from "./BookImage.jsx";

it("renders the BookCoverImage component correctly", () => {
  const url="https://www.penguin.co.uk/content/dam/prh/books/183/183689/9780141199078.jpg.transform/PRHDesktopWide_small/image.jpg";

  const { container } = render(<BookImage url={url} />);
  expect(container).toMatchSnapshot();
});
