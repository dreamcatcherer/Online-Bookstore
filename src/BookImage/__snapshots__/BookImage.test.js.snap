// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the BookCoverImage component correctly 1`] = `
<div>
  <div
    class="imageStyle"
  >
    <img
      src="https://www.penguin.co.uk/content/dam/prh/books/183/183689/9780141199078.jpg.transform/PRHDesktopWide_small/image.jpg"
    />
  </div>
</div>
`;
